pkg/dns/aws/dns.go

```diff
@@ -242,83 +241,102 @@ func (m *Provider) change(record *iov1.DNSRecord, zone configv1.DNSZone, action
 		return fmt.Errorf("unsupported record type %s", record.Spec.RecordType)
 	}
 	// TODO: handle >0 targets
-	domain, target := record.Spec.DNSName, record.Spec.Targets[0]
+	domain, targets := record.Spec.DNSName, record.Spec.Targets
+	updatedRecords := sets.NewString(strings.Join(m.updatedRecords.List(), ","))
 	if len(domain) == 0 {
 		return fmt.Errorf("domain is required")
 	}
-	if len(target) == 0 {
+	if len(targets) == 0 {
 		return fmt.Errorf("target is required")
 	}
 
+	var key string
+	var changes []*route53.Change
+
 	zoneID, err := m.getZoneID(zone)
 	if err != nil {
 		return fmt.Errorf("failed to find hosted zone for record: %v", err)
 	}
 
-	// Find the target hosted zone of the load balancer attached to the service.
-	targetHostedZoneID, err := m.getLBHostedZone(target)
-	if err != nil {
-		return fmt.Errorf("failed to get hosted zone for load balancer target %q: %v", target, err)
-	}
+	for _, target := range targets {
+		// Find the target hosted zone of the load balancer attached to the service.
+		targetHostedZoneID, err := m.getLBHostedZone(target)
+		if err != nil {
+			return fmt.Errorf("failed to get hosted zone for load balancer target %q: %v", target, err)
+		}
+		key = zoneID + domain + target
+		// Configure records and cache updates.
+		// TODO: handle the caching/diff detection in a better way.
+		m.lock.Lock()
+		alreadyHaveRecord := updatedRecords.Has(key)
+		m.lock.Unlock()
```

Comment:
You don't need to lock if you use `updatedRecords` rather than `m.updatedRecords`.

Reply:
done

Reply:
The `m.lock.Lock()` and `m.lock.Unlock()` statements are still here.  Maybe they were re-introduced in a rebase?  The two statements should be deleted.

Actually, you could delete `alreadyHaveRecord := updatedRecords.Has(key)` too (that is, delete all 3 of the lines above) and change the `if alreadyHaveRecord && action == upsertAction {` below to `if updatedRecords.Has(key) && action == upsertAction {`.